test-data/unit/check-classes.test

```diff
@@ -2055,3 +2055,107 @@ class B(object, A): # E: Cannot determine consistent method resolution order (MR
 # flags: --fast-parser
 class C(metaclass=int()):  # E: Dynamic metaclass not supported for 'C'
     pass
+
+[case testVariableSubclass]
+class A:
+    a = 1 # type: int
+class B(A):
+    a = None
+[out]
+
+[case testVariableSubclassAssignMismatch]
+class A:
+    a = 1  # type: int
+class B(A):
+    a = "a"
+[out]
+main: note: In class "B":
+main:4: error: Incompatible types in assignment (expression has type "str", variable has type "int")
+
+[case testVariableSubclassAssignment]
+class A:
+    a = None  # type: int
+class B(A):
+    def __init__(self) -> None:
+        self.a = "a"
+[out]
+main: note: In member "__init__" of class "B":
+main:5: error: Incompatible types in assignment (expression has type "str", variable has type "int")
+
+[case testVariableSubclassTypeOverwrite]
+class A:
+    a = None  # type: int
+class B(A):
+    a = None  # type: str
```

Comment:
Class variables are invariant, so the type should be equivalent (`is_equivalent`) to the base class type. Example:

```
class A:
    a = None  # type: object
    def f(self) -> None:
        self.a = 'x'

class B(A):
    a = None  # type: int
    def g(self) -> None:
        print(self.a + 1)  # ouch, could be str
```